src/Detector/Golang/GolangDetector.cs

```diff
@@ -76,26 +77,23 @@ private string GetVersionFromGoDotMod(DetectorContext context)
                 // go 1.16
 
                 // Regex matching valid version format:
-                //      go 1
                 //      go 1.16
-                //      go 1.16.1
-                Regex regex = new Regex(@"^[\s]*go[\s]+[0-9]+(\.([0-9])+)?(\.([0-9])+)?[\s]*$");
+                //      go 1.16.7
+                Regex regex = new Regex(@"^[\s]*go[\s]+[0-9]+(\.([0-9])+)+[\s]*$");
```

Comment:
nit: add an example of the regex string as comment

Reply:
Updated

Reply:
Updated: Also replaced a + with {1,2} for in regex since previous 1.16.7.8 would succeed. Now That will return null as expected